plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSinkProvider.java

```diff
@@ -124,8 +124,9 @@ public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHa
                         optimizeHandle.getFileFormat(),
                         optimizeHandle.getTableStorageProperties(),
                         maxOpenPartitions);
+            case VACUUM:
+                throw new IllegalArgumentException("VACUUM should not need a page sink");
```

Comment:
just `// handled via ConnectorMedatata.executeTableExecute` and let the `throw` below handle it